internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -488,8 +488,12 @@ func (r *PreprovisioningImageReconciler) getIronicDefaultConfig(log logrus.Field
 
 func (r *PreprovisioningImageReconciler) getIronicConfig(ctx context.Context, log logrus.FieldLogger, infraEnv *aiv1beta1.InfraEnv, infraEnvInternal *common.InfraEnv) (*ICCConfig, error) {
 	var iccConfig *ICCConfig
+	var isOverride bool
```

Comment:
Instead of tracking if this is an override, maybe we can split out the setting of images versus urls.

The problem here is that we don't check the BMO config if we set an image override, but really what we should be doing is getting the URLs from the BMO config, but the image from the override in that case.

I don't see that being fixed in this patch.